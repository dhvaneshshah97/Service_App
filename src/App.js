import React from 'react'
import SimpleCard from './Card/Card';

const App = () => {
  return (
    <div>
      <SimpleCard />
    </div>
  )
}

export default App
