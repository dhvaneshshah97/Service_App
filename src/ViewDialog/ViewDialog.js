import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Paper from '@material-ui/core/Paper';
import Draggable from 'react-draggable';
import { TextField } from '@material-ui/core';

function PaperComponent(props) {
    return (
        <Draggable handle="#draggable-dialog-title" cancel={'[class*="MuiDialogContent-root"]'}>
            <Paper {...props} />
        </Draggable>
    );
}

const ViewDialog = ({ id, name, email, desc, action }) => {
    const [open, setOpen] = useState(false);
    const [values, setValues] = useState({
        name, email, desc
    })

    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    const handleChange = (event) => {
        const [id, value] = event.target;
        setValues({...values, [id]: value})
    }

    return (
        <>
            <Button variant="outlined" color={action === 'update' ? 'inherit' : 'primary'} onClick={handleClickOpen} size="small">
                {action === 'update' ? 'update' : 'view'}
            </Button>
            <Dialog
                open={open}
                onClose={handleClose}
                PaperComponent={PaperComponent}
                aria-labelledby="draggable-dialog-title"
            >
                <DialogTitle style={{ cursor: 'move' }} id="draggable-dialog-title">
                    {name}
                </DialogTitle>
                <DialogContent>

                    <DialogContentText>
                        ID: {id}
                    </DialogContentText>

                    {
                        action === 'view' ? <DialogContentText>
                            Name: {name}
                        </DialogContentText> : <TextField
                            autoFocus
                            margin="dense"
                            id="name"
                            label="Name"
                            fullWidth
                            defaultValue={name}
                            value={values.name}
                            onChange={handleChange}
                        />
                    }

                    {
                        action === 'view' ? <DialogContentText>
                            Email: {email}
                        </DialogContentText> :
                            <TextField
                                autoFocus
                                margin="dense"
                                id="email"
                                label="Email"
                                fullWidth
                                defaultValue={email}
                                value={values.email}
                                onChange={handleChange}
                            />
                    }

                    {
                        action === 'view' ? <DialogContentText>
                            Description: {desc}
                        </DialogContentText> :
                            <TextField
                                autoFocus
                                margin="dense"
                                id="desc"
                                label="Description"
                                fullWidth
                                defaultValue={desc}
                                value={values.desc}
                                onChange={handleChange}
                            />
                    }


                </DialogContent>
                <DialogActions>
                    <Button onClick={handleClose} color="primary">
                        Done
                    </Button>
                </DialogActions>
            </Dialog>
        </>
    );
}

export default ViewDialog;