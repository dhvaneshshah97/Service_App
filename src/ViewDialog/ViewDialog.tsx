import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Paper, { PaperProps } from '@material-ui/core/Paper';
import Draggable from 'react-draggable';
import { TextField } from '@material-ui/core';

interface Props {
    id: number,
    name: string,
    email: string,
    desc: string,
    action: string,
    makeComponentUpdate: Function,
}

function PaperComponent(props: JSX.IntrinsicAttributes & PaperProps) {
    return (
        <Draggable handle="#draggable-dialog-title" cancel={'[class*="MuiDialogContent-root"]'}>
            <Paper {...props} />
        </Draggable>
    );
}

const ViewDialog = ({ id, name, email, desc, action, makeComponentUpdate }: Props): JSX.Element => {
    const [open, setOpen] = useState(false);
    const [values, setValues] = useState({
        id, name, email, desc
    })

    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const name = event.target.name;
        const value = event.target.value;
        setValues({ ...values, [name]: value });
    }

    const handleUpdate = () => {
        let serviceData = JSON.parse(localStorage.getItem('data') || '');
        let idx = serviceData.findIndex((o:{ID:number}) => o.ID === id);
        serviceData[idx].Name = values.name;
        serviceData[idx].Email = values.email;
        serviceData[idx].Description = values.desc;
        localStorage.setItem('data', JSON.stringify(serviceData));
        makeComponentUpdate();
        setOpen(false);
    }

    return (
        <>
            <Button variant="outlined" color={action === 'update' ? 'inherit' : 'primary'} onClick={handleClickOpen} size="small">
                {action === 'update' ? 'update' : 'view'}
            </Button>
            <Dialog
                open={open}
                onClose={handleClose}
                PaperComponent={PaperComponent}
                aria-labelledby="draggable-dialog-title"
            >
                <DialogTitle style={{ cursor: 'move' }} id="draggable-dialog-title">
                    {name}
                </DialogTitle>
                <DialogContent>

                    <DialogContentText>
                        ID: {id}
                    </DialogContentText>

                    {
                        action === 'view' ? <DialogContentText>
                            Name: {name}
                        </DialogContentText> : <TextField
                            autoFocus
                            margin="dense"
                            name="name"
                            label="Name"
                            fullWidth
                            value={values.name}
                            onChange={handleChange}
                        />
                    }

                    {
                        action === 'view' ? <DialogContentText>
                            Email: {email}
                        </DialogContentText> :
                            <TextField
                                autoFocus
                                margin="dense"
                                name="email"
                                label="Email"
                                fullWidth
                                value={values.email}
                                onChange={handleChange}
                            />
                    }

                    {
                        action === 'view' ? <DialogContentText>
                            Description: {desc}
                        </DialogContentText> :
                            <TextField
                                autoFocus
                                margin="dense"
                                name="desc"
                                label="Description"
                                fullWidth
                                value={values.desc}
                                onChange={handleChange}
                                multiline
                                rows={2}
                                maxRows={Infinity}
                            />
                    }


                </DialogContent>
                <DialogActions>
                    {action === 'view' && <Button onClick={handleClose} color="primary">
                        Done
                    </Button>
                    }
                    {
                        action === 'update' && <Button onClick={handleUpdate} color="primary">
                            Update
                        </Button>
                    }


                </DialogActions>
            </Dialog>
        </>
    );
}

export default ViewDialog;