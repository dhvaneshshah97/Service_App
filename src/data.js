export const datasource = [
    {
        ID: '101',
        Name: 'Service-1',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '102',
        Name: 'Service-2',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '103',
        Name: 'Service-3',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '104',
        Name: 'Service-4',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '105',
        Name: 'Service-5',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '106',
        Name: 'Service-6',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '107',
        Name: 'Service-7',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '108',
        Name: 'Service-8',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '109',
        Name: 'Service-9',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '110',
        Name: 'Service-10',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '111',
        Name: 'Service-11',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    },
    {
        ID: '112',
        Name: 'Service-12',
        Email: 'user1@gmail.com',
        Description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'
    }
];