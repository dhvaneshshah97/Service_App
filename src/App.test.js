import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';
import { datasource } from './data';
import renderer from 'react-test-renderer';
import SimpleCard from './Card/Card';
import ViewDialog from './ViewDialog/ViewDialog';

// const mockLocalStorage = () => {
//   const localStorageGetItemMock = jest.spyOn(localStorage.__proto__, 'getItem').mockReturnValue(state);
//   const localStorageRemoveItemMock = jest.spyOn(localStorage.__proto__, 'removeItem');
//   return { localStorageGetItemMock, localStorageRemoveItemMock };
// };


//Snapshot of App component
test('renders app snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});


// View Button
test("View Button Opens Modal correctly", () => {
  render(<SimpleCard
    id={102}
    name="Dhvanesh"
    email="test@gmail.com"
    desc="Description"
  />);
  const button = screen.getByText('view');
  fireEvent.click(button);
  expect(screen.getByText('Done Viewing')).toBeInTheDocument();
});

// Update Button
test("Update Button Opens Modal correctly", () => {
  render(<SimpleCard
    id={102}
    name="Dhvanesh"
    email="test@gmail.com"
    desc="Description"
  />);
  const button = screen.getByText('update');
  fireEvent.click(button);
  expect(screen.getByText('Update Modal')).toBeInTheDocument();
});