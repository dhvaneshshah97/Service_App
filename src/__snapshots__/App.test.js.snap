// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app snapshot 1`] = `
<div
  style={
    Object {
      "margin": "25px",
    }
  }
>
  <h3
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Service Application
  </h3>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  />
  <div
    style={
      Object {
        "margin": "15px 0 5px 0",
        "textAlign": "center",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Reset
      </span>
    </button>
  </div>
</div>
`;
